webapp/src/ts/services/enketo.service.ts

```diff
@@ -435,49 +434,66 @@ export class EnketoService {
     });
   }
 
-  private renderForm(formContext: EnketoFormContext) {
+  private async canAccessForm(formDoc, userContact, instanceData, contactSummary) {
+    const user = userContact || await this.getUserContact();
```

Comment:
I don't think we can call `getUserContact` here since this would prevent admin users (without associated contacts) from opening _contact_ forms.

IMHO, down in the `renderContactForm` we should just add another call to `userContactService.get()` and set the result on the `formContext`. That way, here downstream in the render logic we never have to try getting the user contact again....

Reply:
Looking at the code, this is for any user that doesn't have a contact associated.
Applied suggestion